### PR TITLE
When no cover show title and author

### LINKED ISSUE #2247: Carousels no longer show titles and authors for items without covers (Regression)
### Description

Example URL https://openlibrary.org/subjects/industrial_relations_--_great_britain.

For editions withotu covers, it is impossible to tell what the book is without mouse hovering on any carousel , e.g. Subject carousels, and the 'You might also like' ones.

### Evidence / Screenshot (if possible)
image

### Expectation

The title and author should be shown for books without cover images, similar to how the cover is displayed on the editions details page:
https://openlibrary.org/books/OL22368042M/Employment_relations_in_industrial_society

image

### Details

- **Logged in (Y/N)?** 
- **Browser type/version?** 
- **Operating system?** 

### Stakeholders

@jdlrobson
 @tabshaikh ?
@mekarpeles ?

### PR DESCRIPTION
**Pull request by cdrini**
Closes #2247 (continuation of #2368)

Fix: Works without covers now get an HTML-cover with title/author text.

### Technical
Note this only affects cases where there is no cover; not cases where the cover errors.

### Testing
1. Go to a subjects page e.g. http://localhost:8080/subjects/science
2. Observe that instead of blank covers, covers with text are displayed.

### Evidence
e.g. http://localhost:8080/subjects/science
image

### Stakeholders
@hornc @jdlrobson

### CODE REVIEW

**Comment by cdrini** [comment]
Not sure why it failed; going to try to rerun

**Comment by cdrini** [comment]
Figured out why it failed and fixed it :)

File: openlibrary/templates/books/custom_carousel.html [hunk 1]
```diff
@@ -45,10 +48,16 @@
...
-          $img_attr="$(cover_url)"/>
+        $if cover_url:
+          <img class="bookcover"
+            width="130" height="200"
+            title="$('%s%s'%(title,byline))"
+            $img_attr="$(cover_url)"/>
+        $else:
+          <div class="carousel__item__blankcover">
+            <div class="carousel__item__blankcover--title">$:macros.TruncateString(title, 70)</div>
+            <div class="carousel__item__blankcover--authors">$:macros.TruncateString(', '.join(author_names), 30)</div>
```
Comment (jdlrobson) [inline openlibrary/templates/books/custom_carousel.html]: author_names is not always defined. 

I'm getting
UnboundLocalError: local variable 'author_names' referenced before assignment (falling back to default template)

this needs to be conditional.

Comment (cdrini) [reply openlibrary/templates/books/custom_carousel.html]: Great catch! I'll fix this up :+1:

Comment (cdrini) [reply openlibrary/templates/books/custom_carousel.html]: Done ✅

**Pull request merged**